client/my-sites/themes/index.web.js

```diff
@@ -32,6 +32,13 @@ export default function( router ) {
 				`/design/:vertical(${ verticals })?/:tier(free|premium)?/filter/:filter/:site_id`,
 				validateFilters, siteSelection, singleSite, makeNavigation, makeLayout
 			);
+			if ( config.isEnabled( 'manage/themes/upload' ) ) {
+				router(
+					'/design/upload/:site_id',
+					siteSelection, upload, makeNavigation, makeLayout
+				);
+				// TODO (seear): make `sites` middleware work for single-tree to allow /design/upload
```

Comment:
Hm? Not sure I understand. Can't you just change append a question mark to the route, i.e.
```js
'/design/upload/:site_id?'
````

so it includes `/design/upload`? `siteSelection` should be pretty smart when it comes to handling route fragments.